conda_build/variants.py

```diff
@@ -697,16 +697,18 @@ def get_package_variants(recipedir_or_metadata, config=None, variants=None):
 def get_vars(variants, loop_only=False):
     """For purposes of naming/identifying, provide a way of identifying which variables contribute
     to the matrix dimensionality"""
-    special_keys = {"pin_run_as_build", "zip_keys", "ignore_version"}
-    special_keys.update(set(ensure_list(variants[0].get("extend_keys"))))
+    first_variant, *other_variants = variants
+    special_keys = {
+        "pin_run_as_build",
+        "zip_keys",
+        "ignore_version",
+        *ensure_list(first_variant.get("extend_keys")),
+    }
     loop_vars = [
         k
-        for k in variants[0]
+        for k, v in first_variant.items()
         if k not in special_keys
-        and (
-            not loop_only
-            or any(variant[k] != variants[0][k] for variant in variants[1:])
-        )
+        and (not loop_only or any(variant[k] != v for variant in other_variants))
     ]
```

Comment:
Could this also be written in terms of `set`s?

Reply:
Added a unittest and tinkered some more, see https://github.com/conda/conda-build/compare/6f0474b337583ef357e0244339df116d921940a4..40de0ae1c195ef9c3e6974630d576c96ea64506a